File: Archiver/app_code/clsEXCHANGEHOSTPOP.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql
 
Public Class clsEXCHANGEHOSTPOP

    '** DIM the selected table columns

    Dim DBARCH As New clsDatabaseARCH
    Dim DMA As New clsDma

    Dim UTIL As New clsUtility
    Dim LOG As New clsLogging

    Dim DG As New clsDataGrid

    Dim HostNameIp As String = ""
    Dim UserLoginID As String = ""
    Dim LoginPw As String = ""
    Dim SSL As String = ""
    Dim PortNbr As String = ""
    Dim DeleteAfterDownload As String = ""
    Dim RetentionCode As String = ""
    Dim IMap As String = ""
    Dim Userid As String = ""
    Dim FolderName As String = ""
    Dim LibraryName As String = ""
    Dim isPublic As Boolean = False
    Dim DaysToHold As Integer = 0
    Dim strReject As String = ""
    Dim ckConvertEmlToMsg As Boolean = False

    Dim ConnStr As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID)

    '** Generate the SET methods 
    Public Sub setConvertEmlToMsg(ByRef val As Boolean)
        ckConvertEmlToMsg = val
    End Sub
    Public Sub setReject(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        strReject = val
    End Sub
    Public Sub setDaysToHold(ByRef val As Integer)
        DaysToHold = val
    End Sub
    Public Sub setLibrary(ByRef val As Boolean)
        isPublic = val
    End Sub
    Public Sub setLibrary(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        LibraryName = val
    End Sub
    Public Sub setFolderName(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        FolderName = val
    End Sub

    Public Sub setHostnameip(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Hostnameip' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        HostNameIp = val
    End Sub

    Public Sub setUserloginid(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Userloginid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        UserLoginID = val
    End Sub

    Public Sub setLoginpw(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Loginpw' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        LoginPw = val
    End Sub

    Public Sub setSsl(ByRef val As String)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        SSL = val
    End Sub

    Public Sub setPortnbr(ByRef val As String)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        PortNbr = val
    End Sub

    Public Sub setDeleteafterdownload(ByRef val As String)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        DeleteAfterDownload = val
    End Sub

    Public Sub setRetentioncode(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RetentionCode = val
    End Sub

    Public Sub setImap(ByRef val As String)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        IMap = val
    End Sub

    Public Sub setUserid(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        Userid = val
    End Sub



    '** Generate the GET methods 
    Public Function getFolderName() As String
        Return UTIL.RemoveSingleQuotes(FolderName)
    End Function
    Public Function getHostnameip() As String
        If Len(HostNameIp) = 0 Then
            MessageBox.Show("GET: Field 'Hostnameip' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(HostNameIp)
    End Function

    Public Function getUserloginid() As String
        If Len(UserLoginID) = 0 Then
            MessageBox.Show("GET: Field 'Userloginid' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(UserLoginID)
    End Function

    Public Function getLoginpw() As String
        If Len(LoginPw) = 0 Then
            MessageBox.Show("GET: Field 'Loginpw' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(LoginPw)
    End Function

    Public Function getSsl() As String
        If Len(SSL) = 0 Then
            SSL = "null"
        End If
        Return SSL
    End Function

    Public Function getPortnbr() As String
        If Len(PortNbr) = 0 Then
            PortNbr = "null"
        End If
        Return PortNbr
    End Function

    Public Function getDeleteafterdownload() As String
        If Len(DeleteAfterDownload) = 0 Then
            DeleteAfterDownload = "null"
        End If
        Return DeleteAfterDownload
    End Function

    Public Function getRetentioncode() As String
        Return UTIL.RemoveSingleQuotes(RetentionCode)
    End Function

    Public Function getImap() As String
        If Len(IMap) = 0 Then
            IMap = "null"
        End If
        Return IMap
    End Function

    Public Function getUserid() As String
        Return UTIL.RemoveSingleQuotes(Userid)
    End Function



    '** Generate the Required Fields Validation method 
    Public Function ValidateReqData() As Boolean
        If HostNameIp.Length = 0 Then Return False
        If UserLoginID.Length = 0 Then Return False
        If LoginPw.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the Validation method 
    Public Function ValidateData() As Boolean
        If HostNameIp.Length = 0 Then Return False
        If UserLoginID.Length = 0 Then Return False
        If LoginPw.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the INSERT method 
    Public Function Insert() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""
        s = s + " INSERT INTO ExchangeHostPop("
        s = s + "HostNameIp,"
        s = s + "UserLoginID,"
        s = s + "LoginPw,"
        s = s + "SSL,"
        s = s + "PortNbr,"
        s = s + "DeleteAfterDownload,"
        s = s + "RetentionCode,"
        s = s + "IMap,"
        s = s + "Userid, FolderName, LibraryName, isPublic, DaysToHold, strReject, ConvertEmlToMSG) values ("
        s = s + "'" + HostNameIp + "'" + ","
        s = s + "'" + UserLoginID + "'" + ","
        s = s + "'" + LoginPw + "'" + ","
        s = s + SSL + ","
        s = s + PortNbr + ","
        s = s + DeleteAfterDownload + ","
        s = s + "'" + RetentionCode + "'" + ","
        s = s + IMap + ","
        s = s + "'" + Userid + "'" + ","
        s = s + "'" + FolderName + "'" + ","
        s = s + "'" + LibraryName + "'" + ","
        If isPublic = True Then
            s = s + "1,"
        Else
            s = s + "0,"
        End If
        s = s + DaysToHold.ToString + ", "
        s = s + "'" + strReject + "', "
        If ckConvertEmlToMsg = True Then
            s = s + "1) "
        Else
            s = s + "0) "
        End If
        'log.WriteToArchiveLog("INFO: " + vbCrLf + s)
        Return DBARCH.ExecuteSql(s, ConnStr, False)

    End Function


    '** Generate the UPDATE method 
    Public Function Update(ByVal Host As String, ByVal UserID As String, ByVal UserLogin As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " update ExchangeHostPop set "
        's = s + "HostNameIp = '" + getHostnameip() + "'" + ", "
        's = s + "UserLoginID = '" + getUserloginid() + "'" + ", "
        s = s + "LoginPw = '" + getLoginpw() + "'" + ", "
        s = s + "SSL = " + getSsl() + ", "
        s = s + "PortNbr = " + getPortnbr() + ", "
        s = s + "DeleteAfterDownload = " + getDeleteafterdownload() + ", "
        s = s + "RetentionCode = '" + getRetentioncode() + "'" + ", "
        s = s + "IMap = " + getImap() + ", "
        s = s + "FolderName = '" + Me.getFolderName() + "', "
        s = s + "LibraryName = '" + LibraryName + "', "
        If isPublic = True Then
            s = s + "isPublic = 1, "
        Else
            s = s + "isPublic = 0, "
        End If
        s = s + "DaysToHold = " + DaysToHold.ToString + ", "
        s = s + "strReject = '" + strReject + "', "

        If ckConvertEmlToMsg = True Then
            s = s + "ConvertEmlToMSG = 1 "
        Else
            s = s + "ConvertEmlToMSG = 0 "
        End If

        s = s + " Where "
        s = s + " HostNameIp = '" + Host + "' "
        s = s + " and   Userid = '" + UserID + "'"
        s = s + " and [UserLoginID] = '" + UserLogin + "'"

        Return DBARCH.ExecuteSql(s, ConnStr, False)
    End Function


    '** Generate the SELECT method 
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "HostNameIp,"
        s = s + "UserLoginID,"
        s = s + "LoginPw,"
        s = s + "SSL,"
        s = s + "PortNbr,"
        s = s + "DeleteAfterDownload,"
        s = s + "RetentionCode,"
        s = s + "IMap,"
        s = s + "Userid "
        s = s + " FROM ExchangeHostPop"
        '** s=s+ "ORDERBY xxxx"
        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the Select One Row method 
    Public Function SelectOne(ByVal WhereClause As String) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "HostNameIp,"
        s = s + "UserLoginID,"
        s = s + "LoginPw,"
        s = s + "SSL,"
        s = s + "PortNbr,"
        s = s + "DeleteAfterDownload,"
        s = s + "RetentionCode,"
        s = s + "IMap,"
        s = s + "Userid "
        s = s + " FROM ExchangeHostPop"
        s = s + WhereClause
        '** s=s+ "ORDERBY xxxx"
        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the DELETE method 
    Public Function Delete(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from ExchangeHostPop"
        s = s + WhereClause

        b = DBARCH.ExecuteSql(s, ConnStr, False)
        Return b

    End Function


    '** Generate the Zeroize Table method 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from ExchangeHostPop"

        b = DBARCH.ExecuteSql(s, ConnStr, False)
        Return b

    End Function


    '** Generate Index Queries 
    Public Function cnt_PK_ExchangeHostPop(ByVal HostNameIp As String, ByVal Userid As String, ByVal UserLoginID As String) As Integer

        Dim B As Integer = 0
        Dim TBL As String = "ExchangeHostPop"
        Dim WC As String = "Where HostNameIp = '" + HostNameIp + "' and   Userid = '" + Userid + "' and   UserLoginID = '" + UserLoginID + "'"

        B = DBARCH.iGetRowCount(TBL, WC)

        Return B
    End Function     '** cnt_PK_ExchangeHostPop

    '** Generate Index ROW Queries 
    Public Function getRow_PK_ExchangeHostPop(ByVal HostNameIp As String, ByVal Userid As String, ByVal UserLoginID As String) As SqlDataReader

        Dim rsData As SqlDataReader = Nothing
        Dim TBL As String = "ExchangeHostPop"
        Dim WC As String = "Where HostNameIp = '" + HostNameIp + "' and   Userid = '" + Userid + "' and   UserLoginID = '" + UserLoginID + "'"

        rsData = DBARCH.GetRowByKey(TBL, WC)

        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK_ExchangeHostPop

    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK_ExchangeHostPop(ByVal HostNameIp As String, ByVal Userid As String, ByVal UserLoginID As String) As String

Dim WC AS String  = "Where HostNameIp = '" + HostNameIp + "' and   Userid = '" + Userid + "' and   UserLoginID = '" + UserLoginID + "'" 

        Return WC
    End Function     '** wc_PK_ExchangeHostPop

    '** Generate the SET methods 

End Class

File: Archiver/app_code/clsEXCHANGEHOSTPOP.cs
using global::System.Data.SqlClient;
using System.Windows.Forms;
using global::Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public class clsEXCHANGEHOSTPOP
    {
        public clsEXCHANGEHOSTPOP()
        {
            ConnStr = DBARCH.setConnStr();
        }

        // ** DIM the selected table columns

        private clsDatabaseARCH DBARCH = new clsDatabaseARCH();
        private clsDma DMA = new clsDma();
        private clsUtility UTIL = new clsUtility();
        private clsLogging LOG = new clsLogging();
        private clsDataGrid DG = new clsDataGrid();
        private string HostNameIp = "";
        private string UserLoginID = "";
        private string LoginPw = "";
        private string SSL = "";
        private string PortNbr = "";
        private string DeleteAfterDownload = "";
        private string RetentionCode = "";
        private string IMap = "";
        private string Userid = "";
        private string FolderName = "";
        private string LibraryName = "";
        private bool isPublic = false;
        private int DaysToHold = 0;
        private string strReject = "";
        private bool ckConvertEmlToMsg = false;
        private string ConnStr;     // DBARCH.getGateWayConnStr(gGateWayID)

        // ** Generate the SET methods 
        public void setConvertEmlToMsg(ref bool val)
        {
            ckConvertEmlToMsg = val;
        }

        public void setReject(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            strReject = val;
        }

        public void setDaysToHold(ref int val)
        {
            DaysToHold = val;
        }

        public void setLibrary(ref bool val)
        {
            isPublic = val;
        }

        public void setLibrary(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            LibraryName = val;
        }

        public void setFolderName(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            FolderName = val;
        }

        public void setHostnameip(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Hostnameip' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            HostNameIp = val;
        }

        public void setUserloginid(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Userloginid' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            UserLoginID = val;
        }

        public void setLoginpw(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Loginpw' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            LoginPw = val;
        }

        public void setSsl(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                val = "null";
            }

            val = UTIL.RemoveSingleQuotes(val);
            SSL = val;
        }

        public void setPortnbr(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                val = "null";
            }

            val = UTIL.RemoveSingleQuotes(val);
            PortNbr = val;
        }

        public void setDeleteafterdownload(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                val = "null";
            }

            val = UTIL.RemoveSingleQuotes(val);
            DeleteAfterDownload = val;
        }

        public void setRetentioncode(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            RetentionCode = val;
        }

        public void setImap(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                val = "null";
            }

            val = UTIL.RemoveSingleQuotes(val);
            IMap = val;
        }

        public void setUserid(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            Userid = val;
        }



        // ** Generate the GET methods 
        public string getFolderName()
        {
            return UTIL.RemoveSingleQuotes(FolderName);
        }

        public string getHostnameip()
        {
            if (Strings.Len(HostNameIp) == 0)
            {
                MessageBox.Show("GET: Field 'Hostnameip' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(HostNameIp);
        }

        public string getUserloginid()
        {
            if (Strings.Len(UserLoginID) == 0)
            {
                MessageBox.Show("GET: Field 'Userloginid' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(UserLoginID);
        }

        public string getLoginpw()
        {
            if (Strings.Len(LoginPw) == 0)
            {
                MessageBox.Show("GET: Field 'Loginpw' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(LoginPw);
        }

        public string getSsl()
        {
            if (Strings.Len(SSL) == 0)
            {
                SSL = "null";
            }

            return SSL;
        }

        public string getPortnbr()
        {
            if (Strings.Len(PortNbr) == 0)
            {
                PortNbr = "null";
            }

            return PortNbr;
        }

        public string getDeleteafterdownload()
        {
            if (Strings.Len(DeleteAfterDownload) == 0)
            {
                DeleteAfterDownload = "null";
            }

            return DeleteAfterDownload;
        }

        public string getRetentioncode()
        {
            return UTIL.RemoveSingleQuotes(RetentionCode);
        }

        public string getImap()
        {
            if (Strings.Len(IMap) == 0)
            {
                IMap = "null";
            }

            return IMap;
        }

        public string getUserid()
        {
            return UTIL.RemoveSingleQuotes(Userid);
        }



        // ** Generate the Required Fields Validation method 
        public bool ValidateReqData()
        {
            if (HostNameIp.Length == 0)
                return false;
            if (UserLoginID.Length == 0)
                return false;
            if (LoginPw.Length == 0)
                return false;
            return true;
        }


        // ** Generate the Validation method 
        public bool ValidateData()
        {
            if (HostNameIp.Length == 0)
                return false;
            if (UserLoginID.Length == 0)
                return false;
            if (LoginPw.Length == 0)
                return false;
            return true;
        }


        // ** Generate the INSERT method 
        public bool Insert()
        {
            bool b = false;
            string s = "";
            s = s + " INSERT INTO ExchangeHostPop(";
            s = s + "HostNameIp,";
            s = s + "UserLoginID,";
            s = s + "LoginPw,";
            s = s + "SSL,";
            s = s + "PortNbr,";
            s = s + "DeleteAfterDownload,";
            s = s + "RetentionCode,";
            s = s + "IMap,";
            s = s + "Userid, FolderName, LibraryName, isPublic, DaysToHold, strReject, ConvertEmlToMSG) values (";
            s = s + "'" + HostNameIp + "'" + ",";
            s = s + "'" + UserLoginID + "'" + ",";
            s = s + "'" + LoginPw + "'" + ",";
            s = s + SSL + ",";
            s = s + PortNbr + ",";
            s = s + DeleteAfterDownload + ",";
            s = s + "'" + RetentionCode + "'" + ",";
            s = s + IMap + ",";
            s = s + "'" + Userid + "'" + ",";
            s = s + "'" + FolderName + "'" + ",";
            s = s + "'" + LibraryName + "'" + ",";
            if (isPublic == true)
            {
                s = s + "1,";
            }
            else
            {
                s = s + "0,";
            }

            s = s + DaysToHold.ToString() + ", ";
            s = s + "'" + strReject + "', ";
            if (ckConvertEmlToMsg == true)
            {
                s = s + "1) ";
            }
            else
            {
                s = s + "0) ";
            }
            // log.WriteToArchiveLog("INFO: " + vbCrLf + s)
            return DBARCH.ExecuteSql(s, ConnStr, false);
        }


        // ** Generate the UPDATE method 
        public bool Update(string Host, string UserID, string UserLogin)
        {
            bool b = false;
            string s = "";
            s = s + " update ExchangeHostPop set ";
            // s = s + "HostNameIp = '" + getHostnameip() + "'" + ", "
            // s = s + "UserLoginID = '" + getUserloginid() + "'" + ", "
            s = s + "LoginPw = '" + getLoginpw() + "'" + ", ";
            s = s + "SSL = " + getSsl() + ", ";
            s = s + "PortNbr = " + getPortnbr() + ", ";
            s = s + "DeleteAfterDownload = " + getDeleteafterdownload() + ", ";
            s = s + "RetentionCode = '" + getRetentioncode() + "'" + ", ";
            s = s + "IMap = " + getImap() + ", ";
            s = s + "FolderName = '" + getFolderName() + "', ";
            s = s + "LibraryName = '" + LibraryName + "', ";
            if (isPublic == true)
            {
                s = s + "isPublic = 1, ";
            }
            else
            {
                s = s + "isPublic = 0, ";
            }

            s = s + "DaysToHold = " + DaysToHold.ToString() + ", ";
            s = s + "strReject = '" + strReject + "', ";
            if (ckConvertEmlToMsg == true)
            {
                s = s + "ConvertEmlToMSG = 1 ";
            }
            else
            {
                s = s + "ConvertEmlToMSG = 0 ";
            }

            s = s + " Where ";
            s = s + " HostNameIp = '" + Host + "' ";
            s = s + " and   Userid = '" + UserID + "'";
            s = s + " and [UserLoginID] = '" + UserLogin + "'";
            return DBARCH.ExecuteSql(s, ConnStr, false);
        }


        // ** Generate the SELECT method 
        public SqlDataReader SelectRecs()
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "HostNameIp,";
            s = s + "UserLoginID,";
            s = s + "LoginPw,";
            s = s + "SSL,";
            s = s + "PortNbr,";
            s = s + "DeleteAfterDownload,";
            s = s + "RetentionCode,";
            s = s + "IMap,";
            s = s + "Userid ";
            s = s + " FROM ExchangeHostPop";
            // ** s=s+ "ORDERBY xxxx"
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }


        // ** Generate the Select One Row method 
        public SqlDataReader SelectOne(string WhereClause)
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "HostNameIp,";
            s = s + "UserLoginID,";
            s = s + "LoginPw,";
            s = s + "SSL,";
            s = s + "PortNbr,";
            s = s + "DeleteAfterDownload,";
            s = s + "RetentionCode,";
            s = s + "IMap,";
            s = s + "Userid ";
            s = s + " FROM ExchangeHostPop";
            s = s + WhereClause;
            // ** s=s+ "ORDERBY xxxx"
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }


        // ** Generate the DELETE method 
        public bool Delete(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            WhereClause = " " + WhereClause;
            s = " Delete from ExchangeHostPop";
            s = s + WhereClause;
            b = DBARCH.ExecuteSql(s, ConnStr, false);
            return b;
        }


        // ** Generate the Zeroize Table method 
        public bool Zeroize()
        {
            bool b = false;
            string s = "";
            s = s + " Delete from ExchangeHostPop";
            b = DBARCH.ExecuteSql(s, ConnStr, false);
            return b;
        }


        // ** Generate Index Queries 
        public int cnt_PK_ExchangeHostPop(string HostNameIp, string Userid, string UserLoginID)
        {
            int B = 0;
            string TBL = "ExchangeHostPop";
            string WC = "Where HostNameIp = '" + HostNameIp + "' and   Userid = '" + Userid + "' and   UserLoginID = '" + UserLoginID + "'";
            B = DBARCH.iGetRowCount(TBL, WC);
            return B;
        }     // ** cnt_PK_ExchangeHostPop

        // ** Generate Index ROW Queries 
        public SqlDataReader getRow_PK_ExchangeHostPop(string HostNameIp, string Userid, string UserLoginID)
        {
            SqlDataReader rsData = null;
            string TBL = "ExchangeHostPop";
            string WC = "Where HostNameIp = '" + HostNameIp + "' and   Userid = '" + Userid + "' and   UserLoginID = '" + UserLoginID + "'";
            rsData = DBARCH.GetRowByKey(TBL, WC);
            if (rsData.HasRows)
            {
                return rsData;
            }
            else
            {
                return null;
            }
        }     // ** getRow_PK_ExchangeHostPop

        /// Build Index Where Caluses
        public string wc_PK_ExchangeHostPop(string HostNameIp, string Userid, string UserLoginID)
        {
            string WC = "Where HostNameIp = '" + HostNameIp + "' and   Userid = '" + Userid + "' and   UserLoginID = '" + UserLoginID + "'";
            return WC;
        }     // ** wc_PK_ExchangeHostPop

        // ** Generate the SET methods 

    }
}